tools/serve/test_serve.py

```diff
@@ -8,13 +9,36 @@
 from .serve import Config
 
 
-def test_make_hosts_file():
+@pytest.mark.skipif(platform.uname()[0] == "Windows",
+                    reason="Expected contents are platform-dependent")
+def test_make_hosts_file_nix():
```

Comment:
I think I'd rather just have one function here, with something like:

```
expected = {"",
                          "192.168.42.42\tfoo.bar",
                          "192.168.42.42\tfoo2.bar",
                          "192.168.42.42\twww.foo.bar",
                          "192.168.42.42\twww.foo2.bar",
                          "192.168.42.42\twww1.foo.bar",
                          "192.168.42.42\twww1.foo2.bar",
                          "192.168.42.42\twww2.foo.bar",
                          "192.168.42.42\twww2.foo2.bar",
                          "192.168.42.42\txn--lve-6lad.foo.bar",
                          "192.168.42.42\txn--lve-6lad.foo2.bar",
                          "192.168.42.42\txn--n8j6ds53lwwkrqhv28a.foo.bar",
                          "192.168.42.42\txn--n8j6ds53lwwkrqhv28a.foo2.bar"}
if platform.uname()[0] != "Windows":
    expected ^= {"0.0.0.0\tnonexistent-origin.foo.bar",
                           "0.0.0.0\tnonexistent-origin.foo2.bar"}
```

(except, like, don't have the whitespace a mess from copy/pasting stuff in GitHub :))

Reply:
I considered that, but I chose this organization to simplify the test's semantics. My preference is for tests' expectations to be static so that passing/failing always means the same thing. This has introduced some duplication, though. Is that why you're recommending the tests be consolidated?  Would you be satisfied if we kept both tests but defined the second dict in terms of the first *outside* of the test bodies?